### PR TITLE
#159952005 Users should be able to create and edit their profiles

### PR DESCRIPTION
**Pull request by DrKimpatrick**
## What does this PR do?
Users having their profiles created on registration and edit after.

## Description of Task to be completed?
Have the following endpoints working

`GET /api/profile/user/<username>/`

`PUT /api/profile/update/`

## How should this be manually tested?
Use Postman to do the testing

##### Getting user profile
Accessible to only authorized users, therefore, supply the token in the header.
Insert the username of the person whose profile you want to view

`GET /api/profile/user/<username>/`

#### Update user profile
Accessible to only authorized users, therefore, supply the token in the header.

`PUT /api/user/update/profile/`

Body

   `{
	"profile": {
                           "first_name": "your_first_name",
			  "last_name": "your_last_name",
			  "location": "kampala-uganda",
			  "bio": "About you" 
			}		  
    }`

You can even include `Username, email, and password in the body`

## What are the relevant pivotal tracker stories?
[#159952005](https://www.pivotaltracker.com/story/show/159952005)

### CODE REVIEW

**Comment by huxaiphaer** [comment]
@DrKimpatrick  and @Estaer , edit well your PR 
Below is the area 
```
POST /api/users/login/
`{
"user": {

		  "email": "you@example.com",
		  "password": "your_email"
		}
}`
```

**Comment by huxaiphaer** [comment]
Edit well the comments below also

```
`{
"user": {
"first_name": "your_first_name",
"last_name": "your_last_name",
"location": "kampala-uganda",
"bio": "About you"

		}
}`
```

**Comment by huxaiphaer** [comment]
Please also make the Pivotal Tracker stories a link.

File: authors/apps/authentication/migrations/0001_initial.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Migration(migrations.Migration):
                 ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
                 ('username', models.CharField(db_index=True, max_length=255, unique=True)),
                 ('email', models.EmailField(db_index=True, max_length=254, unique=True)),
-                ('is_active', models.BooleanField(default=False)),
+                ('is_active', models.BooleanField(default=True)),
```
Comment (collinewait) [inline authors/apps/authentication/migrations/0001_initial.py:25]: It had been made False by Esther

Comment (DrKimpatrick) [reply authors/apps/authentication/migrations/0001_initial.py:25]: @collinewait After a rebase, this was changed to True.

File: authors/apps/authentication/serializers.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 from django.shortcuts import get_object_or_404
 
+from authors.apps.profiles.models import UserProfile
```
Comment (collinewait) [inline authors/apps/authentication/serializers.py]: Unused import

Comment (DrKimpatrick) [reply authors/apps/authentication/serializers.py]: It is actually used in `class Meta`

**Pull request merged**